extensions/wikia/Helios/User.class.php

```diff
@@ -226,6 +224,21 @@ public static function setAccessTokenCookie( $accessToken ) {
 		);
 	}
 
+	public static function onUserLogout() {
+		self::invalidateAccessTokenInHelios();
+		self::clearAccessTokenCookie();
+		return true; // So that wfRunHooks evaluates to true.
+	}
+
+	/**
+	 * Call helios invalidate token.
+	 */
+	private static function invalidateAccessTokenInHelios() {
+		$request = \RequestContext::getMain()->getRequest();
+		$heliosClient = self::getHeliosClient();
+		$heliosClient->invalidateToken( self::getAccessToken( $request ) );
+	}
```

Comment:
I think we should put the login/logout and session management in the [auth service](https://github.com/Wikia/app/blob/dev/lib/Wikia/src/Service/User/Auth/AuthService.php). As it is here it's difficult to test. See also the comment below.